pkg/apis/appmesh/v1beta1/types.go

```diff
@@ -453,6 +464,69 @@ type CloudMapServiceStatus struct {
 	NamespaceID *string `json:"namespaceId,omitempty"`
 }
 
+// General TLS Types
+
+type TlsValidationContext struct {
+	Trust TlsValidationContextTrust `json:"trust"`
+}
+
+type TlsValidationContextTrust struct {
+	Acm  *TlsValidationContextAcmTrust  `json:"acm,omitempty"`
+	File *TlsValidationContextFileTrust `json:"acm,omitempty"`
```

Comment:
I figured if someone defined e.g. `File:` but didn't put any content, it shouldn't even be deserialized. Should we pass it through and let the API do the validation? I could follow that, so it's not misleading on what's in the CRD yaml vs what actually gets sent?

Reply:
Yeah I think we want the contract to be as clear as possible in the CRD.

Reply:
To be fair, I don't think this would improve clarity? Unless you mean clarity as in "if you have a stray definition, the API will error at you and point it out".

Anyway, I'll switch it elsewhere too, though I think that would mean we'd want to remove `omitempty` entirely, yes? I won't take that here but hopefully a reviewer closer to this can chime in.

I also just realized I didn't use the `required` filed in the CRD yaml spec, so I gotta add that anyway.

Reply:
Generally `// +optional` and `omitempty` are used together when the field is allowed to be left out.  If the field is required by the App Mesh API, then it shouldn't be omitempty.

Reply:
Okay — I don’t think this is a big deal to keep as-is.